lib/std/Build/Watch.zig

```diff
@@ -811,7 +811,7 @@ const Os = switch (builtin.os.tag) {
 };
 
 pub fn init() !Watch {
-    return Os.init();
+    return if (have_impl) Os.init() else undefined;
```

Comment:
This looks wrong? Did you mean to return an error?

Reply:
Currently build_runner calls this unconditionally and other uses are gated behind `have_impl`, it would probably be better to rework the logic though.

Reply:
As far as I can tell, returning an error will get the correct behavior in the build runner.

Reply:
It should hit this line if have_impl is false: https://github.com/ziglang/zig/blob/c907866d55dd92352e26f7501e5a441210bedaba/lib/compiler/build_runner.zig#L470

Reply:
We know upfront whether `--watch` support is implemented, so checking it in the `rebuild` loop seems a bit silly. I suggest removing that line and just returning something like `error.WatchUnsupportedOnHostOs` from `Watch.init()`, so that this line will bail out early with an error:

https://github.com/ziglang/zig/blob/c907866d55dd92352e26f7501e5a441210bedaba/lib/compiler/build_runner.zig#L418